### PR TITLE
VMware: wait for NSX DV port group readiness

### LINKED ISSUE #13752: VMware NSX network implementation can fail before backing DV port group is visible
### Problem

On VMware environments using the NSX integration, implementing a guest network can fail during `PlugNicCommand` even though `CreateNsxSegmentCommand` has already succeeded.

The NSX segment is created outside vCenter. Its backing distributed virtual port group can therefore take a short time to become visible through the vCenter API. `HypervisorHostHelper.prepareNetwork` currently performs only one immediate lookup for NSX networks and returns failure when that first lookup is empty.

Sanitized management-server sequence:

```text
CreateNsxSegmentCommand ... NsxAnswer result=true
Prepare network on vmwaredvs <redacted>
Failed to create guest network <redacted>
PlugNicAnswer result=false
```

The failure rolls back the NIC attachment and prevents the persistent VPC guest network from being implemented.

### Versions

- Apache CloudStack 4.22.1.0
- VMware vSphere / vCenter 8.0.3
- CloudStack NSX integration

### Steps to reproduce

1. Configure a VMware zone with the CloudStack NSX integration.
2. Create an NSX-backed VPC guest network.
3. Allow the NSX segment command to complete while the backing DV port group is not yet visible through vCenter.
4. Observe the immediate DV port group lookup fail in `HypervisorHostHelper.prepareNetwork`, followed by a failed `PlugNicCommand`.

### Expected behavior

After successful NSX segment creation, CloudStack should use its existing bounded DV port group readiness wait before attempting to attach the NIC. If the port group never becomes visible, the existing timeout should still fail the operation cleanly.

### Proposed fix

Enable the existing `waitForDvPortGroupReady` path for NSX broadcast domains and add a regression test covering an empty first vCenter lookup followed by a successful lookup.

### PR DESCRIPTION
**Pull request by Dogface2k**
### Description

This PR prevents VMware NSX-backed guest network implementation from failing when vCenter has not yet exposed the backing distributed virtual port group immediately after NSX segment creation.

The NSX branch in `HypervisorHostHelper.prepareNetwork` now uses the existing bounded `waitForDvPortGroupReady` path. The change does not retry the CloudStack mutation or create a second network; it only waits for vCenter inventory visibility before returning the already-created port group.

A regression test simulates an empty first vCenter lookup followed by the port group becoming available. The existing immediate-availability behavior remains covered.

Fixes: #13752

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] Build/CI
- [ ] Test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

Not applicable.

### How Has This Been Tested?

Targeted regression and baseline tests were run on the `4.22` branch with Java 17:

```text
HypervisorHostHelperTest#testPrepareNetwork
HypervisorHostHelperTest#testPrepareNetworkWaitsForNsxDvPortGroup

Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
```

The Maven reactor built all 18 required modules successfully, including `cloud-vmware-base`, with zero Checkstyle violations.

### How did you try to break this feature and the system with this change?

The regression test makes the first vCenter DV port group query return no result and the second return the expected port group. The pre-existing immediate-availability test also runs to verify the normal fast path remains unchanged. The production path remains bounded by the existing readiness timeout and returns failure if the port group never appears.

### CODE REVIEW

File: vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java [hunk 1]
```diff
@@ -649,7 +649,8 @@ public static Pair<ManagedObjectReference, String> prepareNetwork(String physica
                 bWaitPortGroupReady = false;
             } else if (BroadcastDomainType.NSX == broadcastDomainType && Objects.nonNull(netName)){
                 networkName = netName;
-                bWaitPortGroupReady = false;
+                // NSX creates the segment outside vCenter, where its backing DV port group may not be immediately visible.
```
Comment (DaanHoogland) [inline vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java]: I don’t think we need this comment, but soit.

```suggestion
```

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan help

**Comment by DaanHoogland** [comment]
@blueorangutan test ol10 vmware-80u3

**Comment by DaanHoogland** [comment]
@blueorangutan test ol9 vmware-80u3

**Review by Pearl1594 — APPROVED** [review]
code lgtm. I don't have a test env to test this fix. But it looks pretty straightforward. Thanks @Dogface2k

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test ol9 vmware-80u3
